### ISSUE TITLE
Struct constructor completion doesn't qualify struct name

### ISSUE THREAD

**Issue by jonas-schievink** [rust-analyzer/rust-analyzer#10771]
```rust
mod submod {
    pub struct Struct {
        pub a: u64,
    }
}

fn f() -> submod::Struct {
    Stru$0
}
```

Triggering completion here and choosing the `Struct {...}` entry results in the following:

```rust
fn f() -> submod::Struct {
    Struct { a: () }
}
```

...which fails to resolve.

**Comment by k-nasa** [comment]
I want to work on it.

**Comment by jonas-schievink** [comment]
@k-nasa I think you can take a look at `Completions::add_qualified_enum_variant` to see how this should be done correctly. `Completions::add_struct_literal` is where the incorrect completion is generated.

**Comment by andrewleverette** [comment]
I added a pull request #10846 that I believe solves this issue. This is my first contribution so any feedback is welcome.

**Comment by Veykril** [comment]
Fixed in #10846
